src/modules/PortAudio/3rdparty/CoreAudio/AudioDevice.h

```diff
@@ -0,0 +1,136 @@
+/*
+     File: AudioDevice.h 
+ Adapted from the CAPlayThough example
+  Version: 1.2.2 
+  
+ Disclaimer: IMPORTANT:  This Apple software is supplied to you by Apple 
+ Inc. ("Apple") in consideration of your agreement to the following 
+ terms, and your use, installation, modification or redistribution of 
+ this Apple software constitutes acceptance of these terms.  If you do 
+ not agree with these terms, please do not use, install, modify or 
+ redistribute this Apple software. 
+  
+ In consideration of your agreement to abide by the following terms, and 
+ subject to these terms, Apple grants you a personal, non-exclusive 
+ license, under Apple's copyrights in this original Apple software (the 
+ "Apple Software"), to use, reproduce, modify and redistribute the Apple 
+ Software, with or without modifications, in source and/or binary forms; 
+ provided that if you redistribute the Apple Software in its entirety and 
+ without modifications, you must retain this notice and the following 
+ text and disclaimers in all such redistributions of the Apple Software. 
+ Neither the name, trademarks, service marks or logos of Apple Inc. may 
+ be used to endorse or promote products derived from the Apple Software 
+ without specific prior written permission from Apple.  Except as 
+ expressly stated in this notice, no other rights or licenses, express or 
+ implied, are granted by Apple herein, including but not limited to any 
+ patent rights that may be infringed by your derivative works or by other 
+ works in which the Apple Software may be incorporated. 
+  
+ The Apple Software is provided by Apple on an "AS IS" basis.  APPLE 
+ MAKES NO WARRANTIES, EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION 
+ THE IMPLIED WARRANTIES OF NON-INFRINGEMENT, MERCHANTABILITY AND FITNESS 
+ FOR A PARTICULAR PURPOSE, REGARDING THE APPLE SOFTWARE OR ITS USE AND 
+ OPERATION ALONE OR IN COMBINATION WITH YOUR PRODUCTS. 
+  
+ IN NO EVENT SHALL APPLE BE LIABLE FOR ANY SPECIAL, INDIRECT, INCIDENTAL 
+ OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF 
+ SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+ INTERRUPTION) ARISING IN ANY WAY OUT OF THE USE, REPRODUCTION, 
+ MODIFICATION AND/OR DISTRIBUTION OF THE APPLE SOFTWARE, HOWEVER CAUSED 
+ AND WHETHER UNDER THEORY OF CONTRACT, TORT (INCLUDING NEGLIGENCE), 
+ STRICT LIABILITY OR OTHERWISE, EVEN IF APPLE HAS BEEN ADVISED OF THE 
+ POSSIBILITY OF SUCH DAMAGE. 
+  
+ Copyright (C) 2013 Apple Inc. All Rights Reserved.
+ Copyright (C) 2017,18 René J.V. Bertin All Rights Reserved.
+  
+*/
+
+#ifndef __AudioDevice_h__
+#define __AudioDevice_h__
+
+#include <CoreServices/CoreServices.h>
+#include <CoreAudio/CoreAudio.h>
+
+#include <qglobal.h>
+
+#ifndef DEPRECATED_LISTENER_API
+#	if !QT_MACOS_PLATFORM_SDK_EQUAL_OR_ABOVE(__MAC_10_11)
```

Comment:
Does not compile:
```
/Users/tomo/Documents/QMPlay2/src/modules/PortAudio/3rdparty/CoreAudio/AudioDevice.h:58: error: token is not a valid binary operator in a preprocessor subexpression
#       if !QT_MACOS_PLATFORM_SDK_EQUAL_OR_ABOVE(__MAC_10_11)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
```

Reply:
What Qt version? I'm not certain if compiling Qt apps with just the CLT is supported officially but if it is that should not make a difference. Those macros are pure Qt but the 10_11 ones were added in June 2015 only.

Does copying the definition for __MAC_10_11 and/or MAC_OS_X_VERSION_10_11 solve the issue (I'm including both but hope only __MAC_10_11 is required)?

Reply:
`QT_MACOS_PLATFORM_SDK_EQUAL_OR_ABOVE` doesn't exist in Qt 5.6, but QMPlay2 must support Qt 5.6. `MAC_OS_X_VERSION_10_11` seems to not be used in project.

Reply:
You defined wrong macro (`QT_OSX_PLATFORM_SDK_EQUAL_OR_ABOVE` != `QT_MACOS_PLATFORM_SDK_EQUAL_OR_ABOVE`).

I think it is better to use `__MAC_OS_X_VERSION_MAX_ALLOWED` directly. Any why there is need to use `DEPRECATED_LISTENER_API`?